packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -153,14 +157,21 @@ export async function askResourceGroupInfo(
     .filter((item) => item.name)
     .map((item) => [item.name, item.location] as [string, string]);
 
-  // TODO: call Azure API directly to list all locations because ARM SDK does not wrap this API.
-  // And then filter by the 'resourceGroup' resource provider.
-  // https://github.com/microsoft/vscode-azuretools/blob/cda6548af53a1c0f538a5ef7542c0eba1d5fa566/ui/src/wizard/LocationListStep.ts#L173
-  const availableLocations = ["East US", "West US"];
+  const locations = await getLocations(ctx, rmClient);
+  if (!locations || locations.length == 0) {
+    return err(
+      returnUserError(
+        new Error(`Failed to list resource group locations`),
+        "Solution",
+        SolutionError.FailedToListResourceGroup
+      )
+    );
+  }
```

Comment:
how about wrapping the error handling logic to the method `getLocations` as well?

Reply:
Fixed.